### PR TITLE
Sanjay

### PR DESCRIPTION
**Pull request by laxmankhanal**
@saugat135 @bishalg @kusalshrestha @anmol-lf 
Here is the initial commit which include the table view included within a scroll view.
Scrolling happens just past the image and then scrolling of the tableView is enabled.

### CODE REVIEW

File: Vaccination/ViewController.swift [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+            "timeRemaining":"2",
+            "vaccinationGiven":"true",
+            "vaccinationAlert":"true"
+        ]
+    ]
+    
+    
+    var tableData = [[String: String]]() // stores the data to be shown in the table view
+    
+    var isTableViewScrolling: Bool = false //flag for the switching between the scroll view and tableview
```
Comment (bishalg) [inline Vaccination/ViewController.swift]: swift will auto cast the variable to Bool when its value is false so no need to write Bool explicitly

File: Vaccination/ViewController.swift [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+    }
+    
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(true)
+        self.vaccinationListTVC.scrollEnabled = false
+        self.vaccinationListTVC.scrollsToTop = false
+    }
+
+    @IBAction func segmentSelected(sender: AnyObject) {
+        if segmentController.selectedSegmentIndex == 0 {
```
Comment (bishalg) [inline Vaccination/ViewController.swift]: make and enum and map it to segment controller and use switch on it

File: Vaccination/ViewController.swift [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+}
+
+extension ViewController: UITableViewDataSource {
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCellWithIdentifier("vaccinationCell") as! VaccinationCell
+        cell.dateLabel.text = self.tableData[indexPath.row]["date"]
+        cell.monthLabel.text = self.tableData[indexPath.row]["month"]
+        cell.vaccinationName.text = self.tableData[indexPath.row]["vaccinationName"]
+        cell.timeRemaining.text = (self.tableData[indexPath.row]["date"])! + " days remaining"
+        return cell
```
Comment (bishalg) [inline Vaccination/ViewController.swift]: make a separate Class named Vaccination and make a method vaccinationForDictionary
then pass tableData to that method and get a Vaccination Object
Optionally, again pass this object to ViewModel which will map the Vaccination object to cell view

File: Vaccination/ViewController.swift [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+        ],
+        [
+            "date":"1",
+            "month":"Jan",
+            "vaccinationName":"BCG",
+            "timeRemaining":"2",
+            "vaccinationGiven":"true",
+            "vaccinationAlert":"true"
+        ]
+    ]
```
Comment (bishalg) [inline Vaccination/ViewController.swift]: separate out all these models to separate file

File: Vaccination/ViewController.swift [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+    }
+    
+    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return tableData.count
+    }
+}
+
+extension ViewController: UITableViewDataSource {
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCellWithIdentifier("vaccinationCell") as! VaccinationCell
```
Comment (bishalg) [inline Vaccination/ViewController.swift]: never use force casting "!" use "If let" or "guard" to check for issue then only use the cell.

File: Vaccination/Assets.xcassets/ic_add_alert_yellow_48dp.imageset/Contents.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "idiom" : "universal",
+      "filename" : "ic_add_alert_yellow_48dp.png",
```
Comment (kusalshrestha) [inline Vaccination/Assets.xcassets/ic_add_alert_yellow_48dp.imageset/Contents.json]: The images in the assets folder has a naming convention.

File: Vaccination/VaccinationCell.swift [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+import UIKit
+
+class VaccinationCell: UITableViewCell {
+
+    @IBOutlet weak var dateLabel: UILabel!
+    @IBOutlet weak var monthLabel: UILabel!
+    @IBOutlet weak var vaccinationName: UILabel!
+    @IBOutlet weak var timeRemaining: UILabel!
+    @IBOutlet weak var vaccinationCheckedImage: UIImageView!
+    @IBOutlet weak var vaccinationNotificationImage: UIImageView!
```
Comment (kusalshrestha) [inline Vaccination/VaccinationCell.swift]: Best practice for IBOutlet is to use Strong

File: Vaccination/MainScrollView.swift [hunk 8]
```diff
@@ -0,0 +1,5 @@
+
+import UIKit
+
+class MainScrollView: UIScrollView {
+}
```
Comment (leapfrog-saugat) [inline Vaccination/MainScrollView.swift]: Add empty line at EOF.

File: Vaccination/VaccinationCell.swift (hunk 7 above)
Comment (leapfrog-saugat) [inline Vaccination/VaccinationCell.swift]: Add space after last line of code in class

File: Vaccination/ViewController.swift [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+    var tableData = [[String: String]]() // stores the data to be shown in the table view
+    
+    var isTableViewScrolling: Bool = false //flag for the switching between the scroll view and tableview
+
+    @IBOutlet weak var coverImage: UIImageView!
+    
+    @IBOutlet weak var mainScrollView: MainScrollView!
+    
+    @IBOutlet weak var segmentController: UISegmentedControl!
+
```
Comment (kusalshrestha) [inline Vaccination/ViewController.swift]: Remove extra spaces

File: Vaccination/ViewController.swift [hunk 10]
```diff
@@ -0,0 +1,287 @@
...
+import UIKit
+
+class ViewController: UIViewController {
+    
+    var dataUpcoming:[[String:String]] = [ //dummy data
```
Comment (leapfrog-saugat) [inline Vaccination/ViewController.swift]: Make a habit of using let for variables which aren't going to be changed.

Comment (leapfrog-saugat) [inline Vaccination/ViewController.swift]: Use appropriate spacing before and after :

File: Vaccination/Child.swift [hunk 11]
```diff
@@ -0,0 +1,31 @@
+
+import Foundation
+
+class Child {
```
Comment (vsujan) [inline Vaccination/Child.swift]: add a line gap after class begins

File: Vaccination/VaccinationListTVC.swift [hunk 12]
```diff
@@ -0,0 +1,4 @@
+import UIKit
+
+class VaccinationListTVC: UITableView {
+}
```
Comment (kusalshrestha) [inline Vaccination/VaccinationListTVC.swift]: add extra line at the EOF

File: Vaccination/Child.swift (hunk 11 above)
Comment (sanjaymhj) [inline Vaccination/Child.swift]: :ok:

File: Vaccination/VaccinationListTVC.swift (hunk 12 above)
Comment (sanjaymhj) [inline Vaccination/VaccinationListTVC.swift]: :ok:

File: Vaccination/ViewController.swift (hunk 2 above)
Comment (sanjaymhj) [inline Vaccination/ViewController.swift]: :ok:

File: Vaccination/ViewController.swift (hunk 1 above)
Comment (sanjaymhj) [inline Vaccination/ViewController.swift]: :ok:

File: Vaccination/ViewController.swift (hunk 3 above)
Comment (sanjaymhj) [inline Vaccination/ViewController.swift]: :ok:

File: Vaccination/ViewController.swift (hunk 4 above)
Comment (sanjaymhj) [inline Vaccination/ViewController.swift]: :ok:

File: Vaccination/ViewController.swift (hunk 5 above)
Comment (sanjaymhj) [inline Vaccination/ViewController.swift]: :ok:

File: Vaccination/VaccinationCell.swift (hunk 7 above)
Comment (sanjaymhj) [inline Vaccination/VaccinationCell.swift]: :ok:

File: Vaccination/MainScrollView.swift (hunk 8 above)
Comment (sanjaymhj) [inline Vaccination/MainScrollView.swift]: :ok:

File: Vaccination/ViewController.swift (hunk 9 above)
Comment (sanjaymhj) [inline Vaccination/ViewController.swift]: :ok:

File: Vaccination/ViewController.swift (hunk 10 above)
Comment (sanjaymhj) [inline Vaccination/ViewController.swift]: :ok:

**Pull request closed without merging**